docs/_content/ontwikkelaars/algemeen/audit-trail.md

```diff
@@ -0,0 +1,248 @@
+2. ...wordt onderdeel van de component API.
+
+We realiseren geen apart component voor de audit trail waar alle componenten
+hun logging heen sturen. Dit zou een synchrone afhankelijkheid creëeren en er
+gaat potentieel veel informatie over het netwerk wat veel overhead veroorzaakt,
+zelfs als niemand deze audit trail ooit raad pleegt. Ten slotte staat deze 
+opzet een eventueel centraal logging component niet in de weg.
+
+De audit trail API wordt dus onderdeel van elk component. Er zijn 2 keuzes, de 
+audit trail API...
+
+1. ...krijgt een zelfstandige API naast de component API 
+   (`example.com/zrc/api/v1` en `example.com/zrc/audittrail/api/v1`).
+2. ...wordt onderdeel van de component API 
+   (`example.com/zrc/api/v1/audittrail`).
+
+Bij het NRC is voor het ontvangen van notificaties door componenten gekozen 
+voor een zelfstandige API. Dit voorkomt dat een wijziging voor de audit trail 
+API alle API's van componenten raakt. Echter, bij het NRC geeft de consumer 
+zelf aan waar deze API te vinden is. Bij de audit trail is hiervan geen sprake
+en dus moet de API consistent en snel te vinden zijn, en het is in essentie ook
+echt een onderdeel van het component.
+
+### Attributen
+
+`uuid` (automatisch)
+
+Unieke identificatie van de audit regel.
+
+`bron` (automatisch)
+
+De naam van get component waar de wijziging in is gedaan. Hoewel dit attribuut 
+in de context van de API niet relevant lijkt, aangezien de API onderdeel is van 
+het component. Mocht deze API echter gebruikt worden in een geaggregeerde 
+context, dan blijft het formaat voor consumers gelijk.
+
+`applicatieId` (verplicht)
+
+Unieke identificatie van de applicatie, binnen de organisatie. Deze komt 
+overeen met het `client_id` attribuut uit de JWT.
+
+`applicatieWeergave` (optioneel)
+
+Vriendelijke naam van de applicatie. Deze kan worden afgeleid uit het `label` 
+attribuut van de `Applicatie` resource in het Atorisatie Component (AC).
+
+`gebruikersId` (verplicht) 
+
+Unieke identificatie van de gebruiker, binnen de organisatie. Deze moet uit de 
+JWT worden gehaald.
+
+*TODO: Moet nog worden toegevoegd aan het JWT en relevante documentatie*
+
+`gebruikersWeergave` (optioneel)
+
+Vriendelijke naam van de gebruiker. Deze moet uit de JWT worden gehaald.
+
+*TODO: Moet nog worden toegevoegd aan het JWT en relevante documentatie*
+
+`actie` (verplicht)
+
+De uitgevoerde handeling *zoals bij notificaties*.
+
+`actieWeergave` (optioneel)
+
+Vriendelijke naam van de actie. Bijvoorbeeld als een relatie tussen 
+een document en een zaak wordt aangemaakt: "gekoppeld aan Zaak MOR-000001".
+
+*TODO: Wellicht ook toevoegen aan het NRC (buiten scope)*
+
+`resultaat` (verplicht)
+
+HTTP status code van de API response van de uitgevoerde handeling. Een HTTP
+status code van "200" of "201" is bijvoorbeeld een succesvol uitgevoerde 
+handeling, terwijl een "403" aangeeft dat de gebruiker of applicatie het wel
+geprobeerd heeft maar de handeling is niet uitgevoerd.
```

Comment:
voor de audit trail zijn mislukte acties niet nodig. Waarom zou je het resultaat dan teruggeven.

Reply:
Begrijp ik punt 4 uit de NORA definitie dan verkeerd?

> De audit trail bevat voldoende gegevens om achteraf te kunnen herleiden (1) 
> welke essentiële handelingen (2) wanneer door (3a) wie of (3b) vanuit welk 
> systeem (4) met welk resultaat zijn uitgevoerd.

Je kan het ook uitleggen als "wat is er gewijzigd" maar dan valt een mislukte wijziging daar dus niet onder?

Reply:
Ja, ik zou het zonder de mislukte acties doen.